Disable cascading writes in ambiguous cases (#3794)
ElektraInitiative/libelektra: doc/news/_preparation_next_release.md

## Highlights

> - `kdb` no longer permits write operations on cascading keys that do not correspond to an existsing key. See the details in the `Tools` section below and the new subsection on cascading writes in the [tutorial](/doc/tutorials/cascading.md) on cascading keys for further information. _(Alexander Firbas)_

Reviewer: double negation

```suggestion
- `kdb` now prohibits write operations on cascading keys that miss a corresponding existing key. See the details in the `Tools` section below and the new subsection on cascading writes in the [tutorial](/doc/tutorials/cascading.md) on cascading keys for further information. _(Alexander Firbas)_
```

- Really add all tools when using `-DTOOLS=ALL`. _(Markus Raab)_
- ZeroMQ Hub: fix compilation and man page. _(Markus Raab)_

> - ambiguous write operations now disabled in kdb _(Alexander Firbas)_

Reviewer: Please give more details here (e.g. that it is about cascading keys, also link to the tutorial.), you can even make a highlight out of this.

Author: I have updated the release notes with further details.

- Really add all tools when using `-DTOOLS=ALL`. _(Markus Raab)_
- ZeroMQ Hub: fix compilation and man page. _(Markus Raab)_

### KDB

- `kdb set`, `kdb meta-set`: Only allow writes to the cascading namespace if the lookup succeeds (Otherwise, the operation is ambiguous and therefore aborted).
  No more guessing of namespaces in case a cascading key is given (`user:, system`: for `kdb set`, `spec:` for `kdb meta-set`), _(Alexander Firbas)_
- `kdb set`, `kdb meta-set`: Validation of keys can no longer be bypassed by using non-cascading keys (except with the new --force (-f) option). _(Alexander Firbas)_
- Disable `-N/--namespace` option in all kdb subcommands _(Alexander Firbas)_

> - Implement new name part getter commands `kdb namespace`, `kdb basename`, `kdb dirname` and `kdb dirkey`. _(Alexander Firbas)_

Reviewer: might need update (dirkey not needed?) Please add other 3 names to doc/help/elektra-glossary.md

Author: The releasenotes got updated; I only added dirname as basename and namespace are already described in that file.
